### PR TITLE
[machine-learning-training] Add sample code for get layer and descrip…

### PR DESCRIPTION
**Pull request by songgot**
…tion for some layer type

- Add sample code for ml_train_model_get_layer()
- Add description for convld, lstmcell, grucell, rnncell and zoneout_lstemcell

Signed-off-by: Hyunil <hyunil46.park@samsung.com>

### CODE REVIEW

**Comment by songgot** [comment]
please review this patch. thank you.

**Review by arisha-srbd — COMMENTED** [review]
Hello @songgot. Hope you are doing well.
I see that you have made some additions to the table, so I wanted to know if you want me to review the whole PR or just check the updated parts. I shall do accordingly. Thank you.

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:187]: okay, thank you.

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:245]: okay, thank you.

**Comment by songgot** [comment]
@arisha-srbd , thank you.
Unfortunately, we do not have permission to this PR.
So Could you merge this PR after review on our side?
I'll let you know after the review is done.

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (jijoongmoon) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:158]: same here for stride

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (jijoongmoon) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:164]: we do not need this.

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (jijoongmoon) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:163]: It needs to be 
```Padding for left, right```

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (jijoongmoon) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:197]: This needs to be removed.

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (jijoongmoon) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:198]: This also needs to be removed. We do not need the time steps in Cell implementation.

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (jijoongmoon) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:212]: This should be removed. we are not going to release the bidirectional LSTM in M1.

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (jijoongmoon) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:610]: add
``` c++
if (status != ML_ERROR_NONE){
   // handle error
}
```

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (jijoongmoon) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:609]: How about giving the notification saying
"layer object owned by model. Do not delete it"

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:157]: thank you.

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:163]: I will fix 'height, width ' to 'left, right'

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:164]: and I will remove this line.

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:197]: okay, I will remove max_timestep

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:198]: okay

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:273]: thank you.

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:609]: okay

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:610]: okay

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (lhs8928) [reply docs/application/native/guides/machine-learning/machine-learning-training.md]: Can we change the description of reset_after to "apply reset gate before/after the matrix multiplication"?

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (jijoongmoon) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:157]: should remove `height, width`. It is single unsigned integer.

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (jijoongmoon) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:158]: same here.

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (jijoongmoon) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:161]: Should remove height/width.

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:157]: thank you

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:158]: thank you

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:161]: okay, thank you

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (lhs8928) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:241]: I guess it's better to remove "Whether"

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (lhs8928) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:244]: same here

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (lhs8928) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:273]: max_timestep, timestep property should be removed from zoneout lstmcell

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (lhs8928) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:272]: test property is not for user. please remove this property

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:273]: okay, thank you.

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:272]: okay, thank you.

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:244]: okay, thank you.

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:241]: okay, thank you.

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:158]: okay, thank you.
Kernel size and Strides look good.

Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:158]: okay, thank you.
Kernel size and Strides look good.

File: docs/application/native/guides/machine-learning/machine-learning-training.md
Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:157]: okay, thank you.

**Review by lhs8928 — APPROVED** [review]
LGTM

**Review by arisha-srbd — COMMENTED** [review]
@songgot hello. Hope you are doing okay. This PR seems approved by @lhs8928. Do you want us to go ahead and merge the PR?

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Comment by songgot** [comment]
hello @arisha-srbd , Could you merge this PR?

**Review by arisha-srbd — APPROVED** [review]
@songgot hello. The PR will be merged today. 

@asm-foysal @mijin85cho I am approving this PR for merging.

**Review by arisha-srbd — COMMENTED** [review]
@songgot hello. Hope you are doing well! 
It seems like this PR contains remarks about Tizen 7.0. Do you think it'd be ok to publish information regarding 7.0 before its release? I should let you know that once this PR will be merged to the master (live) branch, the information would be published in docs.tizen.org. Let me know what you think.

**Pull request merged**
